### DISCUSSION TITLE
Line Highlighting partially broken

### DISCUSSION THREAD

**Discussion by sharpchen** [vuejs/vitepress#4248] [Q&A] [answered]
Line Highlighting partially broken even in a same file.
[Example link](https://sharpchen.github.io/document/Entity%20Framework%20Core/docs/1.%20Schema%20Mapping%20and%20Query%20Control/7.%20Complex%20Property#complex-modeling-across-tables)
I can't reproduce in a minimal project.

The [only section](https://sharpchen.github.io/document/Entity%20Framework%20Core/docs/1.%20Schema%20Mapping%20and%20Query%20Control/7.%20Complex%20Property#complex-modeling-across-tables) got highlighted:
![image](https://github.com/user-attachments/assets/a9c5bfdd-7458-474f-b0ab-96f72a907694)

while others didn't:
![image](https://github.com/user-attachments/assets/401b60cc-df91-4f04-b5b6-f48b90029c0e)

I am not sure how to debug this, all I do is grouping a bunch of files from file system into a multi-sidebar, so they should be compiled into html?

**Comment by brc-dd** [comment]
Ah it's related to https://github.com/vuejs/vitepress/issues/4087

Your theme is splitting `//` and `[!code highlight]` into different tokens. You'll either need to adjust theme to generate same token for comments or use some other theme.

**Reply by sharpchen** [reply]
If that's the case, I don't think the theme is related, should be issue for certain tmLanguage implementation. And this is not always broken, as I paste the file content into a stackbliz vitepress project, everything looks fine.

**Reply by brc-dd** [reply]
tmLanguage is same in both, theme is different. On stackblitz, the default is github-light/dark. In your project you're loading eva-light/dark.

<img width="1084" alt="image" src="https://github.com/user-attachments/assets/42c4a1da-94ff-4a20-b567-901bc784a355">

**Reply by sharpchen** [reply]
Hmm, I thought tokens are generated by tmLanguage, themes only assigns colors for tokens.
These are scopes inspect inside vscode with github dark, it's the same as eva theme I use. I think shiki uses the same source as builtin theme?
![image](https://github.com/user-attachments/assets/3c729eaf-ed3b-4d2d-9bdb-b13a05ba6bb5)
![image](https://github.com/user-attachments/assets/55d89db1-159f-477f-8335-77dbc9208e60)

**Reply by brc-dd** [reply, marked as answer]
Yeah sorry I should've reworded my answer. 

Your theme has different color or style for `"comment"` and `"punctuation.definition.comment"`. This creates two different `<span>`s. Now the way those transformers are written, they only match the regex within same element. This function in shiki core needs to support separate elements - https://github.com/shikijs/shiki/blob/main/packages/transformers/src/utils.ts

with eva dark:

<img width="286" alt="image" src="https://github.com/user-attachments/assets/8bc87406-9826-496c-b4ce-5ff2368c7970">

\
https://github.com/fisheva/Eva-Theme/blob/a90d1a6674271e03c585a0b37c787071e3f01f73/themes/Eva-Dark.json#L930 -- removing `punctuation.definition.comment` from this line will also work.

**Reply by sharpchen** [reply]
I see, `//` is normal while `[!code highglight]` is italic in eva theme, so setting `comment` style to none fixes the problem. Thanks for the instant answer!
